### ISSUE TITLE
vela show failed when definition with complex structure is installed

### ISSUE THREAD

**Issue by Somefive** [oam-dev/kubevela#2036]
**Describe the bug**

Currently, the `vela show --web` will find all definitions in the cluster and render them into Markdown. During the process, it will try to analyze the structure of all fields to generate corresponding descriptions. However, there are cue structures that might be difficult to analyze which lead to rendering failure. For example, the following cue definition is valid and can be used in the KubeVela system normally.
```cue
lifecycle: {
	type: "trait"
	annotations: {}
	labels: {}
	description: "Add lifecycle hooks to workloads."
	attributes: {
		podDisruptive: true
		appliesToWorkloads: ["deployments.apps"]
	}
}
template: {
	patch: spec: template: spec: containers: [{
		lifecycle: {
			if parameter.postStart != _|_ {
				postStart: parameter.postStart
			}
			if parameter.preStop != _|_ {
				preStop: parameter.preStop
			}
		}
	}, ...]
	parameter: {
		postStart?: #LifeCycleHandler
		preStop?:   #LifeCycleHandler
	}
	#Port:             int & >=1 & <=65535
	#LifeCycleHandler: {
		exec: command: [...string]
	} | {
		httpGet: {
			path?:  string
			port:   #Port
			host?:  string
			scheme: *"HTTP" | "HTTPS"
			httpHeaders?: [...{
				name:  string
				value: string
			}]
		}
	} | {
		tcpSocket: {
			port:  #Port
			host?: string
		}
	}
}
```  
The trick part is the `postStart` and `preStart` with type `#LifeCycleHandler` which involves three optional different structures. This will trigger a nil pointer error at `https://github.com/oam-dev/kubevela/blob/2ee3256b20c7993fe24b09e4516f3b2d13b94ea9/references/plugins/references.go#L570` where an error is thrown but not handled.

An alternative code of the above definition could be reached by
```cue
#LifeCycleHandler: {
        exec?: ...
        httpGet?: ...
        tcpSocket?: ...
}
```
Therefore, we can bypass this problem by designing definitions carefully, which in long term, might not be a perfect solution.

**To Reproduce**

Run `vela def apply example.cue` to apply the above definition into the system. It can work with the the [example application](https://github.com/oam-dev/kubevela/blob/724a3ac1da9af4f74df55c9da9d2119f93b5681b/vela-templates/definitions/usage-examples/application-with-lifecycle.yaml), but will make `vela show --web` fail.

**Expected behavior**

`vela show --web` should work normally.

**Screenshots**

**KubeVela Version**

v1.1

**Cluster information**

v1.22.0

**Additional context**

**Comment by wangyuan249** [comment]
Let me provide some other info:
      In current cue parse parameter logic, we can only resolve the single type(int, string and single map), and we need to support complex map-value(struct or list) cue parameter type in future.

![image](https://user-images.githubusercontent.com/35907099/128474524-3d73a14f-f4a6-48ff-a01e-ed3afea54281.png)

**Comment by Somefive** [comment]
@wangyuan249 Yes, these are related problem but not exactly the same. The issue here is to address that in Line#570 there is an error thrown but not handled, which means `subField` will be nil and lead to nil pointer panic when executing `subField.Len()`. This is an unexpected crash for users. We should catch the error, skip the complex definition by now (maybe support in the future), instead of crashing the command.

**Comment by wonderflow** [comment]
fixed by #3039
